src/views/darkroom.c

```diff
@@ -3707,31 +3708,32 @@ int button_pressed(dt_view_t *self,
     dt_control_change_cursor(GDK_HAND1);
     return 1;
   }
-  if(which == 2)
+
+  if(which == 2) // Middle mouse button
   {
     // zoom to 1:1 2:1 and back
-    int procw, proch;
     dt_dev_zoom_t zoom = dt_control_get_dev_zoom();
     int closeup = dt_control_get_dev_closeup();
     float zoom_x = dt_control_get_dev_zoom_x();
     float zoom_y = dt_control_get_dev_zoom_y();
-    dt_dev_get_processed_size(dev, &procw, &proch);
     float scale = dt_dev_get_zoom_scale(dev, zoom, 1<<closeup, 0);
     const float ppd = darktable.gui->ppd;
     const gboolean low_ppd = (darktable.gui->ppd == 1);
-    const float mouse_off_x = x - 0.5f * dev->width;
-    const float mouse_off_y = y - 0.5f * dev->height;
-    zoom_x += mouse_off_x / (procw * scale);
```

Comment:
I think removing this was not expected. At least it breaks the zoom center area. The goal was when zooming to have the same part of the image under the mouse cursor.

Reply:
This is so embarrassing. I didn't catch that when I tested. I know I've screwed up. I am sorry!

I know I can `rebase -i` and delete the commit but I don't know how that's going to work out when the code is already uploaded.

I am sorry I'm taking your time on this, but I don't know how to handle this scenario in git/github with live code. Any suggestions?

Reply:
Don't be sorry ! The good point is that you're getting better at fixing dt :)

For now, just fix the code and push the new commits there. TIA.

Reply:
I 'm still sorry. I don't want to be kicked out. I have learned a lot already and I'll get there :)

I have reverted the commit and I hope it fulfills your request.